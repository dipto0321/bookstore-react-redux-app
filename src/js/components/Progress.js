import React from 'react';
import PropTypes from 'prop-types';
import CircularProgressbar from 'react-circular-progressbar';

const Progress = ({ currentChapter, chapters }) => {
  const percentage = Math.round((currentChapter / chapters) * 100, 2);
  return (
    <div className="progress-body">
      <div className="progress-bar">
        <CircularProgressbar percentage={percentage} />
      </div>
      <div className="progress-percentage">
        <p className="percent-complete">{`${percentage}%`}</p>
        <p className="completed">Completed</p>
      </div>
    </div>
  );
};

Progress.propTypes = {
  currentChapter: PropTypes.number.isRequired,
  chapters: PropTypes.number.isRequired,
};

export default Progress;
