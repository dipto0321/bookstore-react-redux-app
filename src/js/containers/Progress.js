import React from 'react';
import CircularProgressbar from 'react-circular-progressbar';

const Progress = () => {
  const percentage = 66;
  return (
    <div className="progress-body">
      <div className="progress-bar">
        <CircularProgressbar percentage={percentage} />
      </div>
      <div className="progress-percentage">
        <p className="percent-complete">{`${percentage}%`}</p>
        <p className="completed">Completed</p>
      </div>
    </div>
  );
};

export default Progress;
