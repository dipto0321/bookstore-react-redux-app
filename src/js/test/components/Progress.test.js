import React from 'react';
import { shallow } from 'enzyme';
import Progress from '../../components/Progress';

test('Progress component should render correctly', () => {
  const wrapper = shallow(<Progress />);
  expect(wrapper).toMatchSnapshot();
});
