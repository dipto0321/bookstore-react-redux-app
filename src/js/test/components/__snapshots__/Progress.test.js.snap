// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress component should render correctly 1`] = `
<div
  className="progress-body"
>
  <div
    className="progress-bar"
  >
    <CircularProgressbar
      background={false}
      backgroundPadding={0}
      className=""
      classes={
        Object {
          "background": "CircularProgressbar-background",
          "path": "CircularProgressbar-path",
          "root": "CircularProgressbar",
          "text": "CircularProgressbar-text",
          "trail": "CircularProgressbar-trail",
        }
      }
      counterClockwise={false}
      initialAnimation={false}
      percentage={66}
      strokeWidth={8}
      styles={
        Object {
          "background": Object {},
          "path": Object {},
          "root": Object {},
          "text": Object {},
          "trail": Object {},
        }
      }
      text=""
    />
  </div>
  <div
    className="progress-percentage"
  >
    <p
      className="percent-complete"
    >
      66%
    </p>
    <p
      className="completed"
    >
      Completed
    </p>
  </div>
</div>
`;
